monitoring/scripts/track-cid-backfill/src/content.ts

```diff
@@ -0,0 +1,109 @@
+import type { Response } from "node-fetch";
+
+import fetch from "node-fetch";
+import retry from "async-retry";
+
+async function makeRequest(
+  url: string,
+  batch: number[],
+  retries: number,
+  log: boolean,
+  additionalInfo: object
+): Promise<{
+  response?: Response;
+  attemptCount: number;
+  canceled: boolean;
+}> {
+  const additionalInfoMsg = additionalInfo
+    ? ` ${JSON.stringify(additionalInfo)}`
+    : "";
+
+  let attemptCount = 1;
+  console.log(`[DEBUG] making request`);
+  try {
+    const response = await retry(
+      async () =>
+        fetch(url, {
+          method: "POST",
+          headers: {
+            "Content-type": "application/json",
+          },
+          body: JSON.stringify({
+            trackIds: batch,
+          }),
+        }),
+      {
+        retries,
+        factor: 4,
+        minTimeout: 30_000,
+        onRetry: () => {
```

Comment:
no need to change but fyi `onRetry` exposes a counter param

Reply:
This function was copied from network monitoring which was copied from the kpi-dashboard 😂

Reply:
lmao